src/profile-logic/marker-data.js

```diff
@@ -329,7 +329,31 @@ export function filterForMarkerChart(markers: TracingMarker[]) {
   return markers.filter(marker => !isNetworkMarker(marker));
 }
 
-// TODO: add function to merge start and end markers
+export function mergeStartAndEndNetworkMarker(
+  markers: TracingMarker[]
+): TracingMarker[] {
+  const filteredMarkers: TracingMarker[] = [];
+
+  // Flow expects a number instead of a boolead
+  markers.sort((a, b) => {
+    if (a.name < b.name) return -1;
+    if (a.name > b.name) return 1;
+    return 0;
+  });
+
+  for (let i = 0; i < markers.length; i++) {
+    const marker = markers[i];
+    const markerNext = markers[i + 1];
+
+    if (markerNext !== undefined && marker.name === markerNext.name) {
+      continue;
+    }
+    // add marker.data.startTime to marker stop
```

Comment:
What does this comment mean? Is it a TODO? I don't see anything about startTime in the code.

Reply:
The first time we implemented this (before we found out about https://bugzilla.mozilla.org/show_bug.cgi?id=1490963), we took the startTime from the start marker and replaced the startTime in the end marker with it as this is the only value that has a higher importance (e.g. for the queuing start). The others like e.g. priority get updated in the end marker, so we leave them as they are. 

So, this is a todo I wanted to ask you about yesterday, but forgot due to being tired. Sorry.

Do you think we should replace the startTime in the end marker with the one from the start marker?

Reply:
As the startTime for start and end markers differs (and we are refereing to the startTime to show if a request queue started), I'll modify the startTime of the end marker.

See https://github.com/devtools-html/perf.html/issues/1339#issuecomment-429105786 for more info.